packages/replay/src/coreHandlers/handleNetworkBreadcrumbs.ts

```diff
@@ -51,40 +76,80 @@ function _beforeNetworkBreadcrumb(
   }
 
   try {
-    if (breadcrumb.category === 'xhr' && hint && hint.xhr) {
-      _enrichXhrBreadcrumb(
-        breadcrumb as Breadcrumb & { data: object },
-        {
-          xhr: hint.xhr as XMLHttpRequest & SentryWrappedXMLHttpRequest,
-          body: hint.input as RequestBody,
-        },
-        options,
-      );
+    if (_isXhrBreadcrumb(breadcrumb) && _isXhrHint(hint)) {
+      // Enriches the breadcrumb overall
+      _enrichXhrBreadcrumb(breadcrumb, hint, options);
+
+      // Create a replay performance entry from this breadcrumb
+      const result = _makeReplayBreadcrumb('resource.xhr', breadcrumb, hint);
+      addNetworkBreadcrumb(options.replay, result);
     }
 
-    if (breadcrumb.category === 'fetch' && hint) {
-      _enrichFetchBreadcrumb(
-        breadcrumb as Breadcrumb & { data: object },
-        {
-          input: hint.input as HandlerDataFetch['args'],
-          response: hint.response as Response,
-        },
-        options,
-      );
+    if (_isFetchBreadcrumb(breadcrumb) && _isFetchHint(hint)) {
+      // Enriches the breadcrumb overall
+      _enrichFetchBreadcrumb(breadcrumb, hint, options);
+
+      // Create a replay performance entry from this breadcrumb
+      const result = _makeReplayBreadcrumb('resource.fetch', breadcrumb, hint);
+      addNetworkBreadcrumb(options.replay, result);
     }
   } catch (e) {
     __DEBUG_BUILD__ && logger.warn('Error when enriching network breadcrumb');
   }
 }
 
+function _makeReplayBreadcrumb(
```

Comment:
l - just a naming nit (feel free to ignore): Is this function supposed to be used for other breadcrumb types in the future? If that's the case, then the `breadcrumb` and `hint` types are problematic as it constrains it to being a network breadcrumb. Otherwise, let's maybe change the name to `_makeNetworkReplayBreadcrumb`?

Reply:
👍 renaming it!